node/src/components/chainspec_loader.rs

```diff
@@ -456,21 +456,36 @@ impl ChainspecLoader {
             return Effects::new();
         }
 
-        let unplanned_shutdown = match self.next_upgrade {
-            Some(ref next_upgrade) => highest_block_era_id < next_upgrade.activation_point.era_id(),
-            None => true,
+        let (unplanned_shutdown, should_upgrade_if_valid_run) = match self.next_upgrade {
+            Some(ref next_upgrade) => {
+                let unplanned_shutdown =
```

Comment:
`unplanned_shutdown` that we didn't miss an upgrade but the node was restarted anyway, does it?
Would it be more intuitive to use the negation and call it `missed_upgrade`?

Reply:
I'm not sure. In this case, the negation wouldn't quite mean that we missed an upgrade, it would mean that the version that was started is too old (as we already have a block that is after what we consider to be the next upgrade). Which, I guess, is also not quite a "planned shutdown", but I'm not sure what a better name would be :grimacing: